integration/openshift/e2e/runner.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+set -e
+
+GOPATH=${GOPATH:-$(go env GOPATH)}
+script_dir="$(realpath $(dirname $0))"
+source "$script_dir/../../../lib/common.bash"
+tests_dir="$script_dir/../../../"
+source "$script_dir/../lib.sh"
+
+cmd="openshift-tests"
+readonly test_cases_file="${script_dir}/test_cases.txt"
+readonly default_kata_runtimeclass="$kata_runtimeclass"
+
+# Whether to run openshift-tests on dry-run mode or not.
+dry_run="false"
+# Whether to configure the OpenShift cluster or not.
+configure_cluster="false"
+
+function usage() {
+	cat << EOF
+This script runs a sub-set of the OpenShift e2e tests in other to test Kata
+Containers in a running OpenShift cluster.
+
+It assumes that the KUBECONFIG variable is exported and it points to a cluster
+where Kata Containers is already installed.
+
+The cluster also should have:
+ 1) The $default_kata_runtimeclass runtimeClass resource;
+ 2) A running admission controller which annotates any created Pod
+    to use that runtimeClass.
+
+If you choose to have the cluster configured by this script, beware that it
+will not return to its original state on the end.
+
+Usage $0: [-c] [-d] [-r NAME], where:
+  -h, --help   Print this message.
+  -c           Configure the cluster to run the tests.
+  -d           Run the tests in dry-run mode.
+  -r           The runtimeClass name (default is '$default_kata_runtimeclass').
+               This option should not be used with -c.
+EOF
+}
+
+[ -n "$KUBECONFIG" ] || die "The KUBECONFIG variable is unset"
+
+while getopts "hcdr:" opt; do
+	case ${opt} in
+		c)
+			configure_cluster="true"
+			;;
+		d)
+			dry_run="true"
+			;;
+		h)
+			usage
+			exit 0
+			;;
+		r)
+			kata_runtimeclass=${OPTARG}
+			;;
+		*)
+			usage
+			exit 1
+			;;
+	esac
+done
+
+# If the cluster is already configured by the user then only check that the
+# admission controller is active.
+#
+if [ "$configure_cluster" == "true" ]; then
+	[ "$default_kata_runtimeclass" == "$kata_runtimeclass" ] || \
+		die "To configure the cluster use the default runtimeClass"
+	configure_cluster || die "Failed to configure the cluster"
+else
+	info "Check the kata admission controller is active"
+	is_kata_admission_controller_active || \
+		die "Admission controller not active"
+fi
+
+cmd="openshift-tests"
+# Build the openshift tests binary if not available on PATH.
+if ! command -v "$cmd" >/dev/null; then
+	info "'$cmd' command not found on PATH. Build it from source"
+	cmd="$(build_openshift_tests)" || die "The build failed"
+fi
+cmd+=" run"
+
+# Save the tests results in junit format.
+#
+output_dir="${script_dir}/../../../_output/tests/openshift-e2e"
+mkdir -p $output_dir
+cmd+=" --junit-dir ${output_dir}"
+
+if [ $dry_run == "true" ]; then
+	cmd+=" --dry-run"
+fi
+
+cmd+=" -f -"
+info "Run the tests"
+{ set -x; cat $test_cases_file | grep -v ^# | ${cmd}; } || die "FAILED"
```

Comment:
Nit: Quote the grep pattern as GitHub thinks it's a comment :)

Reply:
Even `vim` cannot highlight it properly :D

Reply:
Ouch. It does if you backslash the hash though ;)